packages/docgen/src/engine.js

```diff
@@ -1,25 +1,27 @@
 /**
  * External dependencies.
  */
-const espree = require( 'espree' );
+const babel = require( '@babel/parser' );
 const { flatten } = require( 'lodash' );
 
 /**
  * Internal dependencies.
  */
 const getIntermediateRepresentation = require( './get-intermediate-representation' );
+const babelConfig = require( '../../../babel.config' );
 
-const getAST = ( source ) =>
-	espree.parse( source, {
-		attachComment: true,
-		loc: true,
-		ecmaVersion: 2018,
-		ecmaFeatures: {
-			jsx: true,
-		},
-		sourceType: 'module',
+const getAST = ( source ) => {
+	const { presets, plugins } = babelConfig( {
+		cache: () => {}, // To bypass api.cache is not a function error.
 	} );
 
+	return babel.parse( source || '', {
+		presets,
+		plugins: [ ...plugins, 'jsx' ],
+		sourceType: 'module',
+	} ).program; // unlike espree, the root of babel AST is File, not Program.
```

Comment:
One thing that I usually do in these cases (comments that are useful during the review phase but perhaps don't make a lot of sense in the source code itself) is to add them myself as comments in the PR. Perhaps we could remove it?

Reply:
I'll do it.